### PR TITLE
initial api docs using swagger

### LINKED ISSUE #68: Create documentation for API
In preparation for the private beta, we should add API docs to the repo. Integrate with swagger?

### PR DESCRIPTION
**Pull request by sethvincent**
This pr adds openapi (swagger) docs for all api endpoints.

Closes #68.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# OSM Teams
```
Comment (sethvincent) [inline docs/api.md]: the output of swagger-markdown is not super useful. I wish it included the schemas.

File: docs/api.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        created_at:
+          type: string
+          format: date-time
+        updated_at:
+          type: string
+          format: date-time
+        location:
+          type: string
+          description: geojson point
+    Client:
```
Comment (sethvincent) [inline docs/api.yml]: I'd like to make this more detailed. There might be similar docs somewhere in the hydra project?

**Comment by sethvincent** [comment]
Seeing this error in ci and not sure if I caused it or why the build has been passing previously:

```
{ error: cannot drop extension postgis because other objects depend on it
    at Connection.parseE (/home/circleci/project/node_modules/pg/lib/connection.js:602:11)
    at Connection.parseMessage (/home/circleci/project/node_modules/pg/lib/connection.js:399:19)
    at Socket.<anonymous> (/home/circleci/project/node_modules/pg/lib/connection.js:121:22)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at TCP.onread (net.js:597:20)
  name: 'error',
  length: 311,
  severity: 'ERROR',
  code: '2BP01',
  detail: 'extension postgis_topology depends on extension postgis\nextension postgis_tiger_geocoder depends on extension postgis',
  hint: 'Use DROP ... CASCADE to drop the dependent objects too.',
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'dependency.c',
  line: '992',
  routine: 'reportDependentObjects' }
```

I tried rerunning the workflow but it still failed.

**Comment by kamicut** [comment]
That's weird, I've been seeing that but that's at setup teardown and it's been clearing the tests

File: docs/api.md (hunk 1 above)
Comment (sethvincent) [reply docs/api.md]: haven't found a way to do this yet but can look into it more in the future

File: docs/api.yml (hunk 2 above)
Comment (sethvincent) [reply docs/api.yml]: was able to update this schema based on https://github.com/ory/hydra/blob/master/docs/api.swagger.json#L2365

**Comment by sethvincent** [comment]
that ci error was fixed by #75

File: docs/api.yml [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+          type: integer
+          description: OSM id of user
+        team_id:
+          type: integer
+          description: unique id of the team
+    TeamModeratorList:
+      type: array
+      items:
+        $ref: '#/components/schemas/TeamModerator'
+    TeamFullDetail:
```
Comment (sethvincent) [inline docs/api.yml]: Open to suggestions on the name of this schema. The difference is that when requesting an individual team through `/teams/:id` we get a team object with members and moderators, but when getting a list of teams, the team objects do not include members and moderators. So `TeamFullDetail` includes members & moderators while `Team` does not.

**Pull request merged**